app/uk/gov/hmrc/vatregisteredcompanies/repositories/VatRegisteredCompaniesRepository.scala

```diff
@@ -78,36 +78,23 @@ class   VatRegisteredCompaniesRepository @Inject()(
     logger.info(s"inserting ${entries.length} entries")
     bulkInsert(entries).map(_ => (()))
   }
-
-  private def streamingDelete(deletes: List[VatNumber], payload: PayloadWrapper) = {
-    if (deletes.nonEmpty) {
-      logger.info(s"deleting ${deletes.length} records")
-      val source = Source(deletes)
-      // See https://doc.akka.io/docs/akka/current/stream/operators/Source-or-Flow/throttle.html
-      // we could work out how many records we can safely delete given the size of the collection
-      // and pass that in as a costCalculation
-      val sink = Flow[VatNumber]
-        .throttle(elements, per)
-        .map(vrn => {
-          collection.findAndRemove(Json.obj("vatNumber" -> vrn), None, None, writeConcern = WriteConcern.Default, None, None, Seq.empty).map {
-            _.result[VatNumber]
-          }
-        }).to(Sink.onComplete{x =>
-          x match {
-            case Failure(e) =>
-              logger.error(s"Unable to process streaming deletes at $elements per ${per._1} ${per._2}: ${e.getMessage}")
-            case Success(_) =>
-              bufferRepository.deleteOne(payload)
-              logger.info("Processed streaming deletes")
+  def streamingDelete(deletes: List[VatNumber], payload: PayloadWrapper): Future[Unit] = {
```

Comment:
I'm not sure on this one as Christine wrote it, before it was refactored it was also public. Its also a change not directly related to this ticket but needed to be changed due to bugs.

Reply:
making that function private will not change a thing. it's just being used in this file only that's why i suggested to make it private . generally those function are made public if they are being used anywhere else or needed separate testing. Since this portion is getting tested where we are calling this so make sense to make it private. It's not a big deal, just fpr clean code sanity :)